<launch>

<arg name="chat_room" default="$(optenv ROS_CHAT_ROOM robot_dance_party_0)"/>

  <!-- 
  <node pkg="turtlesim" type ="turtlesim_node" name="sim"/>
  -->
  <node pkg="joy" type="joy_node" name="joystick"/>

  <node pkg="diggy" type="turtle_joy.py" name="robot">
    <remap from="diggy_twist" to="turtle1/cmd_vel"/>
    <remap from="diggy_joy" to="joy_remote"/>
  </node>

  <node pkg="diggy" type="string_joy_map.py" name="joy_to_string">
    <remap from="diggy_joy_in" to="joy"/>
    <remap from="diggy_str_out" to="rtc_offer_node_in"/>
  </node>


  <node pkg="diggy" type="string_joy_map.py" name="string_to_joy">
    <remap from="diggy_joy_out" to="joy_remote"/>
    <remap from="diggy_str_in" to="rtc_answer_node_out"/>
  </node>


  <node pkg="diggy" type="rtc_string_bridge.py" name="rtc_offer_node" output="screen" >
    <remap from="diggy_data_in" to="rtc_offer_node_in"/>
    <remap from="diggy_data_out" to="rtc_offer_node_out"/>
    <param name="role" value="offer"/>
    <param name="room" value="$(arg chat_room)"/>
  </node>

  <node pkg="diggy" type="rtc_string_bridge.py" name="rtc_answer_node" output="screen" >
    <remap from="diggy_data_in" to="rtc_answer_node_in"/>
    <remap from="diggy_data_out" to="rtc_answer_node_out"/>
    <param name="role" value="answer"/>
    <param name="room" value="$(arg chat_room)"/>
  </node>

</launch>
